### ISSUE TITLE
user & role management for Chimera

### ISSUE THREAD

**Issue by marhew** [bptlab/chimera#9]
As an organization, I want to organize my case models and specify who can start cases and execute certain activities, so that I can restrict the number of case workers who can do certain things.
As a case worker, I want to see who has executed a certain task in a case, to be able to ask the right person questions (or to find out who messed everything up). **Therefore, we need a concept for user and role management**

### Concepts

The central notions of our user management sub-system are *user*, *organization*, *(organizational) role*, and *system role* (*organization owner*, *member*, *case owner*, *case participant*, *admin*). Users can be members of ~an organization~ several organizations and hold several organizational roles. Users can create organizations, the creating user has the *(system role) owner*. An owner can define organizational roles, add users to the organization (who then become members of that organization), and assign org. roles to members. Furthermore, users can have the *(system role) admin*, that allows to manage the Chimera system.

Case models belong to organizations. The owner of an organization can specify roles which members need to have in order to view the case model and start cases. The default role necessary for this is *member*, i.e. all members can view the case models of the organization and start cases. The member who created a case becomes the *case owner* of that case. For cases triggered by external events the case start trigger can specify a member or role. Either this member or a random member holding that role will be assigned as case owner. Case owners can invite other members to participate in a case. A member can only become case participant for a case that they are allowed to access the case model of. It should be easily possible to add all members as participants to a case (maybe this should even be the default?). All *case participants* can work on a case.

When creating case models the modeler can specify a role for tasks. This task can only be executed by a member holding that role. If no role is specified, the default role is *member*, i.e. any member who is a case participant can execute the task.
~Modeling roles that can execute certain activities and checking this during execution **is not part of this milestone.** Therefore, the organizational roles might not be necessary at the moment.~
In future roles for tasks should be configurable.

### Features
* create new users, every user is a member of a default organization
* users can log in and out
* create organizations (user)
* create organizational roles (organization owner)
* assign users to organization, they become members (organization owner)
* assign organizational roles to members (organization owner)
* assign case models to organizations
* specify which roles can view case model / start cases
* users only see case models of their organizations
* users only see case models that their role allows
* ~members can only access the cases they own or are a case participant of~
* ~invite members to a case (case owner)~
* ~case knows the members who can access it~
* check user role when executing a task

**Comment by jaSunny** [comment]
Yes, as far as I remember we implemented a User Management System which ist already included.

**Comment by janwestphal306** [comment]
In my opinion, `member` could be an organizational role because it is needed for organizations to decide who is allowed to take part in this specific organization.

**Comment by marhew** [comment]
It is a system role in the sense that the Chimera system needs it to decide which case models to present to a logged-in user (those of the organizations they are a member of).
However, it is also an organization role as you pointed out. In this context, it is the lowest role in (the role lattice of) an organization, used for tasks that anyone, i.e. any member of the organization, can perform. I assume that 'member' is the default role for all tasks that do not specify a role. As such, it is (pre-)defined in all organizations and does not have to be defined by the owner of an organization.
To sum up: 'member' is a system role, because the Chimera frontend needs it, it is pre-defined, and **cannot be added / deleted** like organizational roles.

**Comment by marhew** [comment]
I have removed the part about case ownership and inviting other members of the organization to cases and will put it into another epic.
